panic with unsuported font

### Checklist

- [X] I have searched the [issue tracker](https://github.com/fyne-io/fyne/issues) for open issues that relate to the same problem, before opening a new one.
- [X] This issue only relates to a single bug. I will open new issues for any other problems.

### Describe the bug

Projects that run normally before version 2.3.0 will run incorrectly after compiling with 2.3.0

### How to reproduce

type App struct {
	Proxy.Proxy                  // 继承 代理类
	a                            fyne.App
	w                            fyne.Window
	AddrEntry                    *widget.Entry  // 目标地址
	TimeOutEntry                 *widget.Entry  // 超时设置
	ThreadEntry                  *widget.Entry  // 线程设置
	ProductSelectionMenu         *widget.Select // 支持产品选择框
	VulnerabilitiesSelectionMenu *widget.Select // 漏洞选择框
	UtilizationModeEntry         *widget.Entry  // 利用方式输入框
	DnsLogAddrEntry              *widget.Entry  // DnsLog地址输入框
	StartButton                  *widget.Button // 开始按钮
	StopButton                   *widget.Button // 结束按钮
}

// 全局初始化 基本控件
func (MyApp *App) Init() {
	MyApp.AddrEntry = widget.NewEntry()
	MyApp.ThreadEntry = widget.NewEntry()
	MyApp.ThreadEntry.SetText("1")
	MyApp.TimeOutEntry = widget.NewEntry()
	MyApp.TimeOutEntry.SetText("5")

	MyApp.ProductSelectionMenu = widget.NewSelect([]string{"test"}, func(s string) {

	})

	MyApp.VulnerabilitiesSelectionMenu = widget.NewSelect([]string{"test"}, func(s string) {

	})

	MyApp.UtilizationModeEntry = widget.NewEntry()
	MyApp.DnsLogAddrEntry = widget.NewEntry()

	MyApp.StartButton = widget.NewButtonWithIcon("开始POC", theme.MediaPlayIcon(), func() {
		MyApp.StartButton.Disable()
		MyApp.StopButton.Enable()
	})

	MyApp.StopButton = widget.NewButtonWithIcon("停止POC", theme.MediaStopIcon(), func() {
		MyApp.StopButton.Disable()
		MyApp.StartButton.Enable()
	})
	MyApp.StopButton.Disable()
}


### Screenshots

![11](https://user-images.githubusercontent.com/78897251/217969214-770c1069-8ede-47ed-8bc8-1add9367b95d.PNG)
![12](https://user-images.githubusercontent.com/78897251/217969217-35e62985-74dc-47ef-8547-276046adbb7a.PNG)


### Example code

2023/02/10 08:17:27 Fyne error:  font load error
2023/02/10 08:17:27   Cause: unsupported font format
2023/02/10 08:17:27   At: C:/Users/zhonger/go/pkg/mod/fyne.io/fyne/v2@v2.3.0/internal/painter/font.go:150
panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xc0000005 code=0x0 addr=0x4ec pc=0x7ff7fd7549e0]

goroutine 1 [running, locked to thread]:
github.com/benoitkugler/textlayout/fonts/truetype.(*Font).Upem(0x1?)
        C:/Users/zhonger/go/pkg/mod/github.com/benoitkugler/textlayout@v0.3.0/fonts/truetype/metrics.go:34
github.com/benoitkugler/textlayout/harfbuzz.NewFont({0x7ff7fdf2b078?, 0x0})
        C:/Users/zhonger/go/pkg/mod/github.com/benoitkugler/textlayout@v0.3.0/harfbuzz/fonts.go:98 +0x9c
github.com/go-text/typesetting/shaping.(*HarfbuzzShaper).Shape(0xc0000ef6e0, {{0xc000026770, 0x1, 0x1}, 0x0, 0x1, 0x0, {0x7ff7fdf2b078, 0x0}, 0x340, ...})
        C:/Users/zhonger/go/pkg/mod/github.com/go-text/typesetting@v0.0.0-20230202072710-f146dbf91585/shaping/shaper.go:77 +0x1d6
fyne.io/fyne/v2/internal/painter.walkString({0x7ff7fdf2b078?, 0x0}, {0xc000026074?, 0xc00047a780?}, 0x340, 0x1a4803e3e58?, 0xc0000ef8ac, 0x3f800000, 0x7ff7fde44010)
        C:/Users/zhonger/go/pkg/mod/fyne.io/fyne/v2@v2.3.0/internal/painter/font.go:223 +0x185
fyne.io/fyne/v2/internal/painter.MeasureString({0x7ff7fdf2b078?, 0x0?}, {0xc000026074?, 0x1?}, 0x41500000?, 0x0?)
        C:/Users/zhonger/go/pkg/mod/fyne.io/fyne/v2@v2.3.0/internal/painter/font.go:159 +0x5d
fyne.io/fyne/v2/internal/painter.measureText({0xc000026074, 0x1}, 0x26074?, {0x0, 0x0, 0x0, 0x0, 0x0})
        C:/Users/zhonger/go/pkg/mod/fyne.io/fyne/v2@v2.3.0/internal/painter/font.go:194 +0x8a
fyne.io/fyne/v2/internal/painter.RenderedTextSize({0xc000026074, 0x1}, 0xfd3fcff4?, {0x0, 0x0, 0x0, 0x0, 0x0})
        C:/Users/zhonger/go/pkg/mod/fyne.io/fyne/v2@v2.3.0/internal/painter/font.go:170 +0x9d
fyne.io/fyne/v2/internal/driver/glfw.(*gLDriver).RenderedTextSize(0xc0004b4000?, {0xc000026074?, 0xc00019f9d0?}, 0xfd90d09e?, {0x0, 0x0, 0x0, 0x0, 0x0})
        C:/Users/zhonger/go/pkg/mod/fyne.io/fyne/v2@v2.3.0/internal/driver/glfw/driver.go:73 +0x57
fyne.io/fyne/v2.MeasureText({0xc000026074, 0x1}, 0x1?, {0x0, 0x0, 0x0, 0x0, 0x0})
        C:/Users/zhonger/go/pkg/mod/fyne.io/fyne/v2@v2.3.0/text.go:47 +0x88
fyne.io/fyne/v2/widget.(*RichText).updateRowBounds.func1({0xc000406320?, 0x1, 0x1a4d8beb298?})
        C:/Users/zhonger/go/pkg/mod/fyne.io/fyne/v2@v2.3.0/widget/richtext.go:412 +0x745
        C:/Users/zhonger/go/pkg/mod/fyne.io/fyne/v2@v2.3.0/widget/entry.go:1187 +0x96
fyne.io/fyne/v2/widget.(*BaseWidget).setFieldsAndRefresh(0xc00047a3c0, 0xc00019fe10)
        C:/Users/zhonger/go/pkg/mod/fyne.io/fyne/v2@v2.3.0/widget/widget.go:145 +0x3b
fyne.io/fyne/v2/widget.(*Entry).updateText(0xc00047a3c0, {0x7ff7fdde6f21, 0x1})
        C:/Users/zhonger/go/pkg/mod/fyne.io/fyne/v2@v2.3.0/widget/entry.go:1184 +0x73
fyne.io/fyne/v2/widget.(*Entry).SetText(0xc00047a3c0?, {0x7ff7fdde6f21?, 0xc00047a3c0?})
        C:/Users/zhonger/go/pkg/mod/fyne.io/fyne/v2@v2.3.0/widget/entry.go:470 +0x25
go/ExploitToolsGo/Gui.(*App).Init(0xc000440000)
        D:/go/ExploitToolsGo/Gui/Gui.go:185 +0xfc
go/ExploitToolsGo/Gui.(*App).ShowGui(0xc000440000)
        D:/go/ExploitToolsGo/Gui/Gui.go:330 +0xa5
main.main()
        D:/go/ExploitToolsGo/main.go:34 +0x2a


### Fyne version

2.3.0 -> 2.2.4

### Go compiler version

go 1.19.5

### Operating system

Windows

### Operating system version

windows 10

### Additional Information

_No response_

--- a/internal/painter/font.go
+++ b/internal/painter/font.go
@@ -84,18 +84,33 @@ func CachedFontFace(style fyne.TextStyle, fontDP float32, texScale float32) (fon
 		switch {
 		case style.Monospace:
 			measureFace = loadMeasureFont(theme.TextMonospaceFont())
+			if measureFace == nil {
+				measureFace = loadMeasureFont(theme.DefaultTextMonospaceFont())
+			}
 		case style.Bold:
 			if style.Italic {
 				measureFace = loadMeasureFont(theme.TextBoldItalicFont())
+				if measureFace == nil {
+					measureFace = loadMeasureFont(theme.DefaultTextBoldItalicFont())
+				}
 			} else {
 				measureFace = loadMeasureFont(theme.TextBoldFont())
+				if measureFace == nil {
+					measureFace = loadMeasureFont(theme.DefaultTextBoldFont())
+				}
 			}
 		case style.Italic:
 			measureFace = loadMeasureFont(theme.TextItalicFont())
+			if measureFace == nil {
+				measureFace = loadMeasureFont(theme.DefaultTextItalicFont())
+			}
 		case style.Symbol:
 			measureFace = loadMeasureFont(theme.DefaultSymbolFont())
 		default:
 			measureFace = loadMeasureFont(theme.TextFont())
+			if measureFace == nil {
+				measureFace = loadMeasureFont(theme.DefaultTextFont())
+			}
 		}
 
 		comp.facesMutex.Lock()
@@ -148,6 +163,7 @@ func loadMeasureFont(data fyne.Resource) gotext.Face {
 	loaded, err := gotext.ParseTTF(bytes.NewReader(data.Content()))
 	if err != nil {
 		fyne.LogError("font load error", err)
+		return nil
 	}
 
 	return loaded

